homeassistant/components/imap_email_content/repairs.py

```diff
@@ -0,0 +1,185 @@
+"""Repair flow for imap email content integration."""
+
+from typing import Any
+
+import voluptuous as vol
+import yaml
+
+from homeassistant import data_entry_flow
+from homeassistant.components.imap import DOMAIN as IMAP_DOMAIN
+from homeassistant.components.imap.config_flow import (
+    STEP_USER_DATA_SCHEMA,
+    ConfigFlow as ImapConfigFlow,
+)
+from homeassistant.components.repairs import RepairsFlow
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import (
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+    CONF_VALUE_TEMPLATE,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import issue_registry as ir
+from homeassistant.helpers.typing import ConfigType
+
+from .const import CONF_FOLDER, CONF_SENDERS, CONF_SERVER, DOMAIN
+
+
+@callback
+def process_issue(hass: HomeAssistant, config: ConfigType) -> None:
+    """Register an issue and suggest new config."""
+
+    name: str = config.get(CONF_NAME) or config[CONF_USERNAME]
+
+    issue_id = (
+        f"{name}_{config[CONF_USERNAME]}_{config[CONF_SERVER]}_{config[CONF_FOLDER]}"
+    )
+
+    if CONF_VALUE_TEMPLATE in config:
+        template: str = config[CONF_VALUE_TEMPLATE].template
+        template = template.replace("subject", 'trigger.event.data["subject"]')
+        template = template.replace("from", 'trigger.event.data["sender"]')
+        template = template.replace("date", 'trigger.event.data["date"]')
+        template = template.replace("body", 'trigger.event.data["text"]')
+    else:
+        template = '{{ trigger.event.data["subject"] }}'
+
+    template_sensor_config: ConfigType = {
+        "template": [
+            {
+                "trigger": [
+                    {
+                        "id": "custom_event",
+                        "platform": "event",
+                        "event_type": "imap_content",
+                        "event_data": {"sender": config[CONF_SENDERS][0]},
+                    }
+                ],
+                "sensor": [
+                    {
+                        "state": template,
+                        "name": name,
+                    }
+                ],
+            }
+        ]
+    }
+
+    # Make sure we use the correct config schema
+    data = STEP_USER_DATA_SCHEMA(
+        {
+            CONF_SERVER: config[CONF_SERVER],
+            CONF_PORT: config[CONF_PORT],
+            CONF_USERNAME: config[CONF_USERNAME],
+            CONF_PASSWORD: config[CONF_PASSWORD],
+            CONF_FOLDER: config[CONF_FOLDER],
+        }
+    )
+    try:
+        config_flow = ImapConfigFlow()
```

Comment:
Start an the flow with source import instead like we do when importing config from a platform eg.